internal/provider/data_source_hvn_route.go

```diff
@@ -0,0 +1,102 @@
+package provider
+
+import (
+	"context"
+	"log"
+
+	sharedmodels "github.com/hashicorp/hcp-sdk-go/clients/cloud-shared/v1/models"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-hcp/internal/clients"
+)
+
+func dataSourceHVNRoute() *schema.Resource {
+	return &schema.Resource{
+		Description: "The HVN route data source provides information about an existing HVN route.",
+		ReadContext: dataSourceHVNRouteRead,
+		Timeouts: &schema.ResourceTimeout{
+			Default: &hvnRouteDefaultTimeout,
+		},
+		Schema: map[string]*schema.Schema{
+			// Required inputs
+			"hvn": {
+				Description: "The `self_link` of the HashiCorp Virtual Network (HVN).",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"destination_cidr": {
+				Description:      "The destination CIDR of the HVN route",
+				Type:             schema.TypeString,
+				Required:         true,
+				ValidateDiagFunc: validateIsStartOfPrivateCIDRRange,
```

Comment:
Added validation for `destination_cidr`. Originally I was planning to use the IsCIDR helper as @bcmdarroch had suggested, and use `validation.All` following the example here(https://github.com/hashicorp/terraform-provider-aws/blob/main/aws/resource_aws_eks_cluster.go#L133). However, I went with a custom validator for 2 reasons: 

1-  terraform-provider-hcp only uses `ValidateDiagFunc`, and `validation.All` would not work 
2- `ValidateFunc` is deprecated but most importantly
3- HCP create HVN route endpoint validation (https://github.com/hashicorp/cloud-network/blob/master/service/network/private/service_create_hvn_route.go#L218) uses the validation rule here: https://github.com/hashicorp/cloud-sdk/blob/master/validation/rule_cidrblock.go#L0-L1; this validation is reused in several placess acros HCP endpoints taking a CIDR parameter. `validateIsStartOfPrivateCIDRRange` validates based on the same rules: 

a) CIDR is valid 
b) CIDR is private CIDR block
c) Address is at beginning of CIDR range

Reply:
I would suggest to remove validation here (and probably in other places too), the least duplicated logic we have in Tf provider the easier it will be to maintain. Basic request validation will be done by the backend very quickly and won't impact UX of the TF provider.

Reply:
Thanks for adding client-side validation for this! Looks great, and agreed that this validator should be used for our other CIDRs

Reply:
Oops, just missed @smaant 's response on this. IMO it is reasonable to have this be validated client-side (I'm assuming it is generic and unlikely to change over time), and another slight benefit is that the user will get feedback on the `plan` step instead of seeing it later during `apply`. But otherwise, I do agree that this could be deferred to the backend and the experience would be similar.

Reply:
Replaced the custom validator with `IsCIDR` as we are looking to gain some traction on this work, and merged this PR

Also spoke to provider team (@bcmdarroch & @roaks3 ) as well as some Terraform practitioners; my recommendation is to ensure the custom validator is added back at some point. The aforementioned 3 rules are rudimentary and are unlikely to be removed from the cloud-sdk validation package, hence drift is less of a concern. If the validation rules in cloud-sdk were to change, we are unlikely to remove rules, and only likely to add additional rules, in which case the custom validator covering 3 existing rules will be a good idea in terms of improving the practitioner experience for this provider.